src/reporter.ts

```diff
@@ -0,0 +1,292 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as os from 'os';
+import SauceLabs from 'saucelabs';
+import { TestRun, Suite, Test, Status as SauceStatus } from '@saucelabs/sauce-json-reporter'
+import { SummaryFormatter, formatterHelpers, Status, IFormatterOptions } from '@cucumber/cucumber'
+import { IParsedTestStep } from '@cucumber/cucumber/lib/formatter/helpers/test_case_attempt_parser';
+import { FormatOptions } from '@cucumber/cucumber/lib/formatter';
+import { Envelope } from '@cucumber/messages';
+
+type SauceRegion = 'us-west-1' | 'eu-central-1' | 'staging';
+
+type ReportsRequestBody = {
+  name?: string;
+  browserName?: string;
+  browserVersion?: string;
+  platformName?: string;
+  framework?: string;
+  frameworkVersion?: string;
+  passed?: boolean;
+  startTime: string;
+  endTime: string;
+  build?: string;
+  tags?: string[];
+  suite?: string;
+};
+
+type Asset = {
+  filename: string;
+  data: Buffer;
+};
+
+class SauceReporter extends SummaryFormatter {
+  testRun: TestRun;
+  suiteName: string;
+  browserName: string;
+  build: string;
+  tags: string[];
+  region: SauceRegion;
+  outputFile: string;
+  shouldUpload: boolean;
+  assets: Asset[];
+  api?: SauceLabs;
+  cucumberVersion: string;
+  startedAt?: Date;
+  endedAt?: Date;
+  videoStartTime?: number;
+  consoleLog: string[];
+  passed: boolean;
+
+  constructor(config: IFormatterOptions) {
+    super(config)
+
+    const reporterConfig: FormatOptions = config.parsedArgvOptions;
```

Comment:
I don't think you need to cast this to `FormatOptions`. That type is not technically public so you have to import from `../lib/...` which is a code smell in and of itself.

Second, the `FormatOptions` type allows for arbitrary keys:
```export interface FormatOptions {
  colorsEnabled?: boolean
  rerun?: FormatRerunOptions
  snippetInterface?: SnippetInterface
  snippetSyntax?: string
  printAttachments?: boolean
  [customKey: string]: any
}```

The customKey bit there. 

So you can just access `reporterConfig.whatever`. May have to explicitly type the property accesses though